bolt/include/bolt/Core/BinaryContext.h

```diff
@@ -1363,6 +1363,17 @@ class BinaryContext {
     if (std::optional<uint32_t> Size = MIB->getSize(Inst))
       return *Size;
 
+    // Pseudo instrs will not be emiitted and have no size.
+    if (MIB->isPseudo(Inst)) {
+      return 0;
+    }
+
+    // Directly return 4 because AArch64 instructions always have a
+    // fixed size of 4 bytes.
+    if (isAArch64()) {
```

Comment:
What if we define this method in AArch64MCPBuilder.cpp ?

Reply:
You mean define `getSize ` in `AArch64MCPlusBuilder`, and in `computeInstructionSize`, 

```
if (isAArch64()) { 
   return MIB->getSize(Inst)
}
```

Am I right?

Reply:
no, inside AArch64MCPlusBuilder.cpp

  std::optional<uint32_t> getSize(const MCInst &Inst) const override {
       return 4;
}

Reply:
I think the default implementation should return 0 and we proceed with the encodeInstruction below in such case.

Reply:
ok~

Reply:
What about now?

Reply:
Thank you! This function will cost a lot time in AArch64 for big binaries.

Reply:
I think the way it is now implemented interference with the idea of getting instruction size in MIB with annotations. We first want to check annotations and then check with target specific API. I don't mind if getSIze would call target-specific function in case of annotation absense that would return optional value too. But don't fully replace annotation version getSize. Although currently it look like it is OK for aarch64 (only looks, since with annotations we can do other funny things potentionaly) we might meet such problems in another architectures, so everything must my in sync.

Reply:
You mean don't override `getSize ` in  AArch64MCPlusBuilder but add some logic in computeInstructionSize to check an AArch64 instruction

Reply:
I mean that "getSize" declaration should be final. And first of all we should check annotations in getSize. But if no annotation presented we should call getInstructionSize()/getTargetSize()/etc that would be target-specific directly from getSize() that would return  the same optional uint32_t. Default implementation of getInstructionSize() should return  std::nullopt.

Reply:
ok